src/v/cluster/partition_allocation_strategies.h

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2020 Vectorized, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#pragma once
+
+#include "cluster/partition_allocator.h"
+#include "cluster/types.h"
+#include "model/metadata.h"
+
+namespace cluster {
+
+class round_robin_allocation_strategy final : public allocation_strategy::impl {
```

Comment:
Why is this public tho. Why not simply expose it as a generator func like the record-batch-reader

Reply:
Also why is the interface not defined in its own file with comments on the base class interface. 

For example, this is missing a key component which are constraints 

or a set of labels to allocate.

Reply:
Think about the k8s min-max allocator . The interface is simple
Enough there and we could tweak this to add a future api later. As is the separation isn’t as impactful